pkgs/test/lib/src/_package_hybrid_test_helper.dart

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
```

Comment:
We don't want testing code in our `lib` directory, even if under `lib/_src`.

If this is shared across packages we can either create multiple copies or create a shared testing package which is not published in the mono_repo (if one does not already exist).

Reply:
Although for this one it looks like it could be created as a part of the package being tested? (using the test descriptor apis)

Reply:
You could add a single new package in `integration_tests`, which contains both the test and the code to spawn (under lib of that package, it isn't a published package its just for testing purposes).

I would probably just go that route as it would be easiest. You could move this entire test into there, which might make it easier to understand as well (and stop using test descriptor in this package).